library/MedEx/API.php

```diff
@@ -0,0 +1,2092 @@
+            sqlStatement($sqlFLOW,array($data['msg_type'],$data['pc_eid']));//if it is not in tracker what will happen?  Error and continue?
+            $log['sql']=$sqlFlow;
+            $this->MedEx->logging->log_this($log);     
+        } elseif ($data['msg_reply']=="CALL") {
+            $sqlUPDATE = "UPDATE openemr_postcalendar_events set pc_apptstatus = 'CALL' where pc_eid=?";
+            $test = sqlQuery($sqlUPDATE,array($data['pc_eid']));
+            $log['sql']=$sqlUPDATE;
+            $this->MedEx->logging->log_this($log);
+        } elseif (($data['msg_type']=="AVM") && ($data['reply']=="STOP")) {
+            //if reply = "STOP" update patient demographics to disallow this mode of communication
+            //well this is where we are also for RECALLS
+            if (!$data['patient_id']) {
+                if ($data['e_pid']) {
+                    $data['patient_id'] = $patient['e_pid'];
+                } else {
+                    $query = "SELECT * from openemr_postcalendar_events where pc_eid=?"; //assume one patient per appointment pc_eid/slot...
+                    $patient = sqlFetchArray(sqlStatement($query,array($data['pc_eid'])));  //otherwise this will need to be a loop          
+                    $data['patient_id'] = $patient['pid'];
+                }
+            }
+            $sqlUPDATE = "UPDATE patient_data set hipaa_voice = 'NO' where pid=?";
+            sqlQuery($sqlUPDATE,array($data['patient_id']));
+        } elseif (($data['msg_type']=="SMS") && ($data['reply']=="STOP")) {
+            $sqlUPDATE = "UPDATE patient_data set hipaa_allowsms = 'NO' where pid=?";
+            $log['sql']=$sqlUPDATE;
+            $this->MedEx->logging->log_this($log);
+            sqlQuery($sqlUPDATE,array($data['e_pid']));
+        } elseif (($data['msg_type']=="EMAIL") && ($data['reply']=="STOP")) {
+            $sqlUPDATE = "UPDATE patient_data set hipaa_allowemail = 'NO' where pid=?";
+            $log['sql']=$sqlUPDATE;
+            $this->MedEx->logging->log_this($log);
+            sqlQuery($sqlUPDATE,array($data['patient_id']));
+        } 
+        if (($data['msg_reply']=="SENT")||($data['reply']=="READ")) {
+            $sqlDELETE = "DELETE FROM medex_outgoing where msg_pc_eid=? and msg_reply='To Send'";
+            $log['sql']=$sqlDELETE;
+            $this->MedEx->logging->log_this($log);
+            sqlQuery($sqlDELETE,array($data['pc_eid']));
+        }
+        //process E-MAIL responses
+        //TODO        --- return manual processed records.
+
+        $response['comments'] = $data['pc_eid']." - ".$data['campaign_uid']." - ".$data['msg_type']." - ".$data['reply']." - ".$data['extra'];
+        $response['pid'] = $data['pid'];
+        $response['success'] = $data['msg_type']." reply";
+        
+        return $response;
+    }
+}
+
+class Logging extends base {
+    public function log_this($data) {
+        //look in the openEMR files to figure out how to log this.
```

Comment:
No idea what to do with logging.  You'll see I struggle with this question throughout the code.   I presume someone has a notion regarding this.  The message trail is actually kept in the DB table medex_outgoing so except for debugging I don't think it is really needed.  OK /tmp it is...